go/tasks/plugins/webapi/bigquery/plugin.go

```diff
@@ -0,0 +1,503 @@
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhaseRetryableFailure, systemExecutionError, taskInfo)
+
+	// This error returns when you refer to a resource (a dataset, a table, or a job) that doesn't exist.
+	// This can also occur when using snapshot decorators to refer to deleted tables that have recently been
+	// streamed to.
+	case "notFound":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhasePermanentFailure, userExecutionError, taskInfo)
+
+	// This job error returns when you try to access a feature that isn't implemented.
+	case "notImplemented":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhasePermanentFailure, userExecutionError, taskInfo)
+
+	// This error returns when your project exceeds a BigQuery quota, a custom quota, or when you haven't set up
+	// billing and you have exceeded the free tier for queries.
+	case "quotaExceeded":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhaseRetryableFailure, userExecutionError, taskInfo)
+
+	case "rateLimitExceeded":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhaseRetryableFailure, userExecutionError, taskInfo)
+
+	// This error returns when you try to delete a dataset that contains tables or when you try to delete a job
+	// that is currently running.
+	case "resourceInUse":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhaseRetryableFailure, systemExecutionError, taskInfo)
+
+	// This error returns when your query uses too many resources.
+	case "resourcesExceeded":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhasePermanentFailure, userExecutionError, taskInfo)
+
+	// This error returns when your query's results are larger than the maximum response size. Some queries execute
+	// in multiple stages, and this error returns when any stage returns a response size that is too large, even if
+	// the final result is smaller than the maximum. This error commonly returns when queries use an ORDER BY
+	// clause.
+	case "responseTooLarge":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhasePermanentFailure, userExecutionError, taskInfo)
+
+	// This status code returns when a job is canceled.
+	case "stopped":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhasePermanentFailure, userExecutionError, taskInfo)
+
+	// Certain BigQuery tables are backed by data managed by other Google product teams. This error indicates that
+	// one of these tables is unavailable.
+	case "tableUnavailable":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhaseRetryableFailure, systemExecutionError, taskInfo)
+
+	// The job timed out.
+	case "timeout":
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhasePermanentFailure, userExecutionError, taskInfo)
+
+	default:
+		return pluginsCore.PhaseInfoFailed(pluginsCore.PhasePermanentFailure, systemExecutionError, taskInfo)
+	}
+}
+
+func createTaskInfo(resourceMeta *ResourceMetaWrapper) *core.TaskInfo {
+	timeNow := time.Now()
+	j := fmt.Sprintf("bq:%s:%s", resourceMeta.JobReference.Location, resourceMeta.JobReference.JobId)
```

Comment:
Can you also move this to a function with a name representing what it's used for